activation/post.go

```diff
@@ -90,15 +90,15 @@ type PostSetupOpts struct {
 }
 
 type PostProviderID struct {
-	value *int64
+	InnerValue *int64
```

Comment:
This was to allow for serialisation/serialisation of the config in order to deep copy it in the test. Being able to serialize/deserialize config seems like it could  be a useful property to have.

Reply:
Our config is (de-)serializable using `viper`. This change would introduce an option to set an illegal ID as provider:

This config will cause the node to fail at startup with the error that `-2` is not a legal provider ID:

```json
{
    "smeshing-opts-provider": -2
}
```

With the change this is possible:

```json
{
    "smeshing-opts-provider": {
        "InnerValue": -2
    }
}
```

Reply:
Yes I think it is possible. So I'll revert this modification and not copy the config.